src/input_buttons_desktop.cpp

```diff
@@ -21,130 +21,133 @@
-	buttons[LEFT].push_back(Keys::JOY_4);
-	buttons[RIGHT].push_back(Keys::JOY_6);
-	buttons[DECISION].push_back(Keys::JOY_1);
-	buttons[CANCEL].push_back(Keys::JOY_3);
-	buttons[SHIFT].push_back(Keys::JOY_5);
-	buttons[N0].push_back(Keys::JOY_10);
-	buttons[N1].push_back(Keys::JOY_11);
-	buttons[N2].push_back(Keys::JOY_12);
-	buttons[N3].push_back(Keys::JOY_13);
-	buttons[N4].push_back(Keys::JOY_14);
-	buttons[N5].push_back(Keys::JOY_15);
-	buttons[N6].push_back(Keys::JOY_16);
-	buttons[N7].push_back(Keys::JOY_17);
-	buttons[N8].push_back(Keys::JOY_18);
-	buttons[N9].push_back(Keys::JOY_19);
-	buttons[PLUS].push_back(Keys::JOY_20);
-	buttons[MINUS].push_back(Keys::JOY_21);
-	buttons[MULTIPLY].push_back(Keys::JOY_22);
-	buttons[DIVIDE].push_back(Keys::JOY_23);
-	buttons[PERIOD].push_back(Keys::JOY_24);
-	buttons[DEBUG_MENU].push_back(Keys::JOY_7);
-	buttons[DEBUG_THROUGH].push_back(Keys::JOY_9);
+		// FIXME: Random joystick keys mapping, better to read joystick configuration from .ini
+		,{UP, Keys::JOY_8}
+		,{DOWN, Keys::JOY_2}
+		,{LEFT, Keys::JOY_4}
+		,{RIGHT, Keys::JOY_6}
+		,{DECISION, Keys::JOY_1}
+		,{CANCEL, Keys::JOY_3}
+		,{SHIFT, Keys::JOY_5}
+		,{N0, Keys::JOY_10}
+		,{N1, Keys::JOY_11}
+		,{N2, Keys::JOY_12}
+		,{N3, Keys::JOY_13}
+		,{N4, Keys::JOY_14}
+		,{N5, Keys::JOY_15}
+		,{N6, Keys::JOY_16}
+		,{N7, Keys::JOY_17}
+		,{N8, Keys::JOY_18}
+		,{N9, Keys::JOY_19}
+		,{PLUS, Keys::JOY_20}
+		,{MINUS, Keys::JOY_21}
+		,{MULTIPLY, Keys::JOY_22}
+		,{DIVIDE, Keys::JOY_23}
+		,{PERIOD, Keys::JOY_24}
+		,{DEBUG_MENU, Keys::JOY_7}
+		,{DEBUG_THROUGH, Keys::JOY_9}
 #endif
 
 #if defined(USE_JOYSTICK_HAT)  && defined(SUPPORT_JOYSTICK_HAT)
-	buttons[DOWN].push_back(Keys::JOY_HAT_DOWN);
-	buttons[LEFT].push_back(Keys::JOY_HAT_LEFT);
-	buttons[RIGHT].push_back(Keys::JOY_HAT_RIGHT);
-	buttons[UP].push_back(Keys::JOY_HAT_UP);
+		,{DOWN, Keys::JOY_HAT_DOWN}
+		,{LEFT, Keys::JOY_HAT_LEFT}
+		,{RIGHT, Keys::JOY_HAT_RIGHT}
+		,{UP, Keys::JOY_HAT_UP}
 
 #endif
 
 #if defined(USE_JOYSTICK_AXIS)  && defined(SUPPORT_JOYSTICK_AXIS)
-	buttons[LEFT].push_back(Keys::JOY_AXIS_X_LEFT);
-	buttons[RIGHT].push_back(Keys::JOY_AXIS_X_RIGHT);
-	buttons[DOWN].push_back(Keys::JOY_AXIS_Y_DOWN);
-	buttons[UP].push_back(Keys::JOY_AXIS_Y_UP);
+		,{LEFT, Keys::JOY_AXIS_X_LEFT}
+		,{RIGHT, Keys::JOY_AXIS_X_RIGHT}
+		,{DOWN, Keys::JOY_AXIS_Y_DOWN}
+		,{UP, Keys::JOY_AXIS_Y_UP}
 #endif
 
 #if defined(USE_TOUCH) && defined(SUPPORT_TOUCH)
-	buttons[DECISION].push_back(Keys::ONE_FINGER);
-	buttons[CANCEL].push_back(Keys::TWO_FINGERS);
+		,{DECISION, Keys::ONE_FINGER}
+		,{CANCEL, Keys::TWO_FINGERS}
 #endif
+	};
+}
 
-	dir_buttons.resize(10);
-	dir_buttons[2].push_back(DOWN);
-	dir_buttons[4].push_back(LEFT);
-	dir_buttons[6].push_back(RIGHT);
-	dir_buttons[8].push_back(UP);
+Input::DirectionMappingArray Input::GetDefaultDirectionMappings() {
+	return {
+		{ Direction::DOWN, DOWN },
+		{ Direction::LEFT, LEFT },
+		{ Direction::RIGHT, RIGHT },
+		{ Direction::UP, UP },
```

Comment:
Is there a reason for using leading commas? Except here btw, here they are trailing.

Reply:
Yes, with leading commas you don't need to worry about the last element being different. This makes it easier handling all these `#ifdef`